src/Components/Admin/EnvironmentEditor/environment.js

```diff
@@ -282,8 +286,39 @@ export default function EnvironmentEditor({
                         file_type: transitions[i].fileType,
                         type: "image",
                         file_content: transitions[i].file,
+                        s3Key: "",
                     };
 
+                    if (
+                        Object.prototype.hasOwnProperty.call(
+                            transitions[i],
+                            "imageSrc"
+                        ) &&
+                        transitions[i].imageSrc !== ""
+                    ) {
+                        const oldS3Key = transitions[i].imageSrc.replace(
+                            process.env.REACT_APP_ADMIN_ASSET_PREFIX,
+                            ""
+                        );
+                        const new_file_type_len =
+                            transitions[i].fileType.length;
+                        const old_file_type = oldS3Key.slice(
+                            oldS3Key.length - new_file_type_len
+                        );
+
+                        if (old_file_type === transitions[i].fileType) {
+                            body.s3Key = oldS3Key;
+                        } else {
+                            const imageList = [oldS3Key];
+                            await deleteTransitionImages(
+                                {
+                                    scenarioId: environmentId,
+                                    imagesList: imageList,
+                                },
+                                handleError
+                            );
+                        }
+                    }
```

Comment:
I may need this logic explained to me in more detail.

My understanding:
If image exists, find the old S3 key and the old file type (so this can be the S3 key and .jpg let's say).
Issue: what if I delete the image/transition? the transitions imageSrc will be empty, so this conditional isn't run, and no file deletion is happening on S3 side right? Unless I am missing something.

Next, if file type is the same, set s3Key to the old s3Key. What happens now? Are we uploading to AWS the new image, and reusing the key value?

If file type is not the same, we delete image. What happens here? Do we still upload a new image if I have selected a new image?

Reply:
So there was some caching stuff going on with using the same link. What we now do to tackle this is delete the image regardless and upload the new image to a new key.

We maintain a list of images to delete and then at the end when we click submit, the images on that list are executed 😵😵. Lmk if you have questions about the new approach.